src/MetalPerformanceShaders/MPSDefs.cs

```diff
@@ -217,5 +263,132 @@ public struct MPSImageKeypointRangeInfo {
 		public nuint MaximumKeypoints;
 		public float MinimumThresholdValue;
 	}
+
+	[TV (11, 3), iOS (11, 3), Mac (10, 13, 4, onlyOn64: true)]
+	public struct MPSStateTextureInfo {
+		public nuint Width;
+		public nuint Height;
+		public nuint Depth;
+		public nuint ArrayLength;
+
+#pragma warning disable 0169 // Avoid warning when building core.dll and the unused reserved fields
+		nuint _PixelFormat;
+		nuint _TextureType;
+		nuint _TextureUsage;
+
+		//NSUInteger _reserved [4];
+		nuint Reserved0;
+		nuint Reserved1;
+		nuint Reserved2;
+		nuint Reserved3;
+#pragma warning restore 0169
+#if !COREBUILD
+		public MTLPixelFormat PixelFormat {
+			get => (MTLPixelFormat) (ulong) _PixelFormat;
+			set => _PixelFormat = (nuint) (ulong) value;
+		}
+
+		public MTLTextureType TextureType {
+			get => (MTLTextureType) (ulong) _TextureType;
+			set => _TextureType = (nuint) (ulong) value;
+		}
+
+		public MTLTextureUsage TextureUsage {
+			get => (MTLTextureUsage) (ulong) _TextureUsage;
+			set => _TextureUsage = (nuint) (ulong) value;
+		}
+#endif
+	}
+	[TV (11, 3), iOS (11, 3), Mac (10, 13, 4, onlyOn64: true)]
+	[Native]
+	public enum MPSStateResourceType : ulong {
+		None = 0,
+		Buffer = 1,
+		Texture = 2,
+	}
+
+	[TV (12, 0), Mac (10, 14, onlyOn64: true), iOS (12, 0)]
+	[Native]
+	public enum MPSIntersectionType : ulong {
+		Nearest = 0,
+		Any = 1,
+	}
+
+	[TV (12, 0), Mac (10, 14, onlyOn64: true), iOS (12, 0)]
+	[Native]
+	public enum MPSTriangleIntersectionTestType : ulong {
+		Default = 0,
+		Watertight = 1,
+	}
+
+	[TV (12, 0), Mac (10, 14, onlyOn64: true), iOS (12, 0)]
+	[Native]
+	public enum MPSBoundingBoxIntersectionTestType : ulong {
+		Default = 0,
+		AxisAligned = 1,
+	}
+
+	[TV (12, 0), Mac (10, 14, onlyOn64: true), iOS (12, 0)]
+	[Native]
+	public enum MPSRayMaskOptions : ulong {
```

Comment:
the name of the `enum` makes it sounds it's a `[Flag]` ?
also `None` instead of `Default` also _kind of_ points to a `[Flag]`